core/include/usFramework.h

```diff
@@ -0,0 +1,151 @@
+ * used to manage and control the created framework instance.
+ */
+class US_Core_EXPORT Framework : public Module
+{
+public:
+    Framework(void);
+    ~Framework(void);
+
+    /**
+     * Initialize this Framework.
+     *
+     * \throw BundleException If this Framework could not be initialized.
+     */
+    void init(void);
+
+    /**
+     * \param timeout Maximum number of milliseconds to wait until this Framework has completely stopped. A value of
+     * zero will wait indefinitely.
+     * 
+     * \remarks Wait until this Framework has completely stopped. The stop and update methods on a Framework performs an 
+     * asynchronous stop of the Framework. This method can be used to wait until the asynchronous stop of this 
+     * Framework has completed. This method will only wait if called when this Framework is in the 
+     * STARTING, ACTIVE, or STOPPING states. Otherwise it will return immediately. A Framework Event is returned 
+     * to indicate why this Framework has stopped.
+     *
+     * \return A Framework Event indicating the reason this method returned. The following FrameworkEvent types 
+     * may be returned by this method.
+     * - STOPPED - This Framework has been stopped.
+     * - STOPPED_UPDATE - This Framework has been updated which has shutdown and will now restart.
+     * - ERROR - The Framework encountered an error while shutting down or an error has occurred which forced the framework to shutdown.
+     * - WAIT_TIMEDOUT - This method has timed out and returned before this Framework has stopped.
+     * 
+     * \throw InterruptedException If another thread interrupted the current thread before or while the current
+     *  thread was waiting for this Framework to completely stop. The interrupted status of the current thread 
+     *  is cleared when this exception is thrown.
+     * \throw IllegalArgumentException If the value of timeout is negative.
+     */
+    void waitForStop(void) { /* TODO: implement */ }
+
+    void Start();
+    void Stop() { /* TODO: implement */ }
+    void Update() { /* TODO: implement */ }
+    void Uninstall();
+
+    std::string GetLocation() const;
+
+    /****************************************************
+     * BEGIN - Non OSGi compliant functions
+     *
+     * These functions originally came from the ModuleSettings class, 
+     * and are duplicated here temporarily to transition functionality 
+     * until a solution is agreed upon.
+     ****************************************************/
+
+    /**
+     * Enable or disable auto-loading support.
+     *
+     * \param enable If \c true, enable auto-loading support, disable it otherwise.
+     *
+     * \remarks Calling this method will have no effect if support for
+     * auto-loading has not been configured into the CppMicroServices library of it
+     * it has been disabled by defining the US_DISABLE_AUTOLOADING envrionment variable.
+     */
+    void SetAutoLoadingEnabled(bool enable);
```

Comment:
In the interim, we should keep auto-install/load, until it has been factored out into a service.